File: backend/src/models/entities/conversation.py
import uuid

from sqlalchemy import String, Text, ForeignKey, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.models.entities.document import Base


class Conversation(Base):
    __tablename__ = "conversations"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    title: Mapped[str] = mapped_column(String(255), default="新对话")
    created_at: Mapped[str] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )

    messages: Mapped[list["Message"]] = relationship(
        "Message", back_populates="conversation",
        order_by="Message.created_at", cascade="all, delete-orphan",
    )


class Message(Base):
    __tablename__ = "messages"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    conversation_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("conversations.id", ondelete="CASCADE"), index=True,
    )
    role: Mapped[str] = mapped_column(String(20))  # "user" | "assistant"
    content: Mapped[str] = mapped_column(Text)
    retrievals_json: Mapped[str | None] = mapped_column(Text)
    created_at: Mapped[str] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )

    conversation: Mapped["Conversation"] = relationship(
        "Conversation", back_populates="messages",
    )

File: frontend/src/apis/conversation.ts
import api from './index'

export function listConversations() {
  return api.get('/conversations') as Promise<{
    items: { id: string; title: string; created_at: string }[]
    total: number
  }>
}

export function getConversation(id: string) {
  return api.get(`/conversations/${id}`) as Promise<{
    id: string
    title: string
    messages: {
      id: string
      role: string
      content: string
      retrievals: { tool_name: string; content_preview: string }[]
      created_at: string
    }[]
    created_at: string
  }>
}

export function deleteConversation(id: string) {
  return api.delete(`/conversations/${id}`)
}
